src/index.ts

```diff
@@ -257,17 +265,33 @@ function compareAsc(a: unknown, b: unknown) {
   } else if (b === null) {
```

Comment:
For the number, string and boolean you created methods that will determine the return value, why didn't you do that for the `a===0`? I think that one is more complex than the `compareAsNumber` method you extracted now. I think you should either extract all of them (except the -1, that would make no sense), or extract none. I think there wasn't anything wrong with the one liners in the previous version.

Reply:
I extracted some functions to resolve a cognitive complexity code smell. If you think that this doesn't reduce the complexity then I'm fine with reverting the change.

Reply:
I believe sonar. He's way smarter than me. Thanks for the clarification.